### PR TITLE
EVGo - Refactor to DictParser and yield item properly

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Fix https://github.com/alltheplaces/alltheplaces/issues/9656 ?

### CODE REVIEW

**Comment by iandees** [comment]
This seems to be missing a bunch of chargers: https://evgo.com/find-a-charger/

**Comment by CloCkWeRX** [comment]
Screenshot_20240820-064301

**Comment by CloCkWeRX** [comment]
UI says "search by plug share" - https://www.plugshare.com/ - perhaps they are migrating?

**Comment by iandees** [comment]
8 is a bit too far away from the list of chargers they have on their website. I don't see anything about plugshare on the page I linked to. There's a big 'ol JSON blob with what looks like a hard to parse list of all ~200 charging stations on the map.
